crates/viewer/re_view_spatial/src/transform_cache.rs

```diff
@@ -392,127 +476,173 @@ impl TransformCacheStoreSubscriber {
-                .invalidated_transforms
-                .push(InvalidatedTransforms {
-                    entity_path: entity_path.clone(),
-                    times: vec![TimeInt::STATIC],
-                    aspects,
-                });
+        let entity_path = chunk.entity_path();
 
-            // Adding a static transform invalidates ALL times for this entity on ALL timelines, since the resulting transforms at all times may be different now.
-            // Furthermore, since we want to incorporate the static transforms into all timelines, we have to add this event to all timelines.
-            for (timeline, per_timeline_transforms) in &mut self.per_timeline {
-                let entity_transforms = per_timeline_transforms
-                    .per_entity
-                    .entry(entity_path.hash())
-                    .or_insert_with(|| {
-                        // Need to add an entry now if there wasn't one before.
-                        // Also note that the static transforms we use to construct this might touch on aspects that aren't invalidated, so it's still important to pass that in.
-                        TransformsForEntity::new(
-                            Some(*timeline),
-                            self.static_timeline.per_entity.get(&entity_path.hash()),
-                        )
-                    });
-                if aspects.contains(TransformAspect::Tree) {
-                    per_timeline_transforms
-                        .invalidated_transforms
-                        .push(InvalidatedTransforms {
-                            entity_path: entity_path.clone(),
-                            times: std::iter::once(TimeInt::STATIC)
-                                .chain(entity_transforms.tree_transforms.keys().copied())
-                                .collect(),
-                            aspects,
-                        });
+        for (timeline, time_column) in chunk.timelines() {
+            let per_timeline = self.per_timeline.entry(*timeline).or_insert_with(|| {
+                CachedTransformsForTimeline::new(timeline, &self.static_timeline)
+            });
+
+            // All of these require complex latest-at queries that would require a lot more context,
+            // are fairly expensive, and may depend on other components that may come in at the same time.
+            // (we could inject that here, but it's not entirely straight forward).
+            // So instead, we note down that the caches are invalidated for the given entity & times.
+
+            // This invalidates any time _after_ the first event in this chunk.
+            // (e.g. if a rotation is added prior to translations later on,
+            // then the resulting transforms at those translations change as well for latest-at queries)
+
+            let mut invalidated_times = Vec::new();
+            let Some(min_time) = time_column.times().min() else {
+                continue;
+            };
```

Comment:
Technically, this is too conservative (i.e. you'd want a bunch of `min_time_for_component` instead).
In practice, this is absolutely fine and is probably not worth the extra complexity.